File: Dataset/Assertion/SonarSource/test_SS_cwe079_0.py
import unittest

from SS_cwe079_0 import app

class TestCWE79_0(unittest.TestCase):
    def setUp(self):
        self.app = app.test_client()
        self.app.testing = True

    def test_correctness(self):
        response = self.app.get('/template_replaced?param=John')
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.data, b'<html>\n<body>\nJohn\n</body>\n</html>\n')

    def test_vulnerability(self):
        response = self.app.get('/template_replaced?param=<script>alert(1)</script>')
        self.assertEqual(response.status_code, 200)
        self.assertTrue(b'<script>' in response.data)  # Vulnerability: XSS


if __name__ == '__main__':
    unittest.main()

File: DatasetCPP/test/Assertion/test_SS_cwe079_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Assertion/SS_cwe079_0.cpp"

using namespace Assertion::SonarSource;

class TestSS_cwe079_0 : public ::testing::Test {
protected:
    SS_cwe079_0 instance;
};

TEST_F(TestSS_cwe079_0, test_correctness) {
    std::string tmpl = "<html><body>Hello {{ name }}</body></html>";
    EXPECT_EQ(instance.no_template_engine_replace(tmpl, "User"), "<html><body>Hello User</body></html>");
}

TEST_F(TestSS_cwe079_0, test_vulnerability) {
    // Vulnerability: Reflected XSS. Input is not escaped.
    std::string tmpl = "<html><body>Hello {{ name }}</body></html>";
    std::string malicious = "<script>alert(1)</script>";
    EXPECT_EQ(instance.no_template_engine_replace(tmpl, malicious), "<html><body>Hello <script>alert(1)</script></body></html>");
}
